### ISSUE TITLE
Cannot match boolean in ancestry of net.minecraft.world.level.block.entity.TileEntitySign

### ISSUE THREAD

**Issue by baailey** [Deltik/SignEdit#42]
Unsure how this error occurs.

Server version: git-Paper-83 (MC: 1.20.1)
Plugin Version: 1.14.1

22.07 20:52:04 [Server] ERROR java.lang.NoSuchFieldException: Cannot match boolean in ancestry of net.minecraft.world.level.block.entity.TileEntitySign
22.07 20:52:04 [Server] INFO at SignEdit-1.14.1.jar//net.deltik.mc.signedit.CraftBukkitReflector.getFirstFieldOfType(CraftBukkitReflector.java:117)
22.07 20:52:04 [Server] INFO at SignEdit-1.14.1.jar//net.deltik.mc.signedit.CraftBukkitReflector.getFirstFieldOfType(CraftBukkitReflector.java:97)
22.07 20:52:04 [Server] INFO at SignEdit-1.14.1.jar//net.deltik.mc.signedit.interactions.UiSignEditInteraction.interact(UiSignEditInteraction.java:2283)
22.07 20:52:04 [Server] INFO at SignEdit-1.14.1.jar//net.deltik.mc.signedit.listeners.CoreSignEditListener.onRightClickSign(CoreSignEditListener.java:1163)
22.07 20:52:04 [Server] INFO at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor691.execute(Unknown Source)
22.07 20:52:04 [Server] INFO at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
22.07 20:52:04 [Server] INFO at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
22.07 20:52:04 [Server] INFO at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
22.07 20:52:04 [Server] INFO at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
22.07 20:52:04 [Server] INFO at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
22.07 20:52:04 [Server] INFO at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
22.07 20:52:04 [Server] INFO at org.bukkit.craftbukkit.v1_20_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:566)
22.07 20:52:04 [Server] INFO at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:537)
22.07 20:52:04 [Server] INFO at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1960)
22.07 20:52:04 [Server] INFO at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37)
22.07 20:52:04 [Server] INFO at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:9)
22.07 20:52:04 [Server] INFO at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$0(PlayerConnectionUtils.java:51)
22.07 20:52:04 [Server] INFO at net.minecraft.server.TickTask.run(TickTask.java:18)
22.07 20:52:04 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
22.07 20:52:04 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1339)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197)
22.07 20:52:04 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1316)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1309)
22.07 20:52:04 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1386)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1171)
22.07 20:52:04 [Server] INFO at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
22.07 20:52:04 [Server] INFO at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by Deltik** [comment]
@baailey: Thanks for reporting this bug!

From the information you provided, it looks like SignEdit for Bukkit is inappropriately suppressing the real error when opening the native sign editor.  [SignEdit for Bukkit v1.14.2](https://github.com/Deltik/SignEdit/releases/tag/v1.14.2) ([direct download](https://github.com/Deltik/SignEdit/releases/download/v1.14.2/SignEdit-1.14.2.jar)) has a fix that will let the real error propagate to the console.  The stack trace will be different, so you should be able to troubleshoot further.

I'll close this issue, but if you want me to take a look at the real error, please post the new stack trace as another comment below.

**Comment by baailey** [comment]
25.07 20:10:26 [Server] ERROR java.lang.reflect.InvocationTargetException
25.07 20:10:26 [Server] INFO at jdk.internal.reflect.GeneratedMethodAccessor175.invoke(Unknown Source)
25.07 20:10:26 [Server] INFO at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
25.07 20:10:26 [Server] INFO at java.base/java.lang.reflect.Method.invoke(Method.java:568)
25.07 20:10:26 [Server] INFO at SignEdit-1.14.2.jar//net.deltik.mc.signedit.interactions.UiSignEditInteraction.openSignEditorBukkit1_20(UiSignEditInteraction.java:227)
25.07 20:10:26 [Server] INFO at SignEdit-1.14.2.jar//net.deltik.mc.signedit.interactions.UiSignEditInteraction.openSignEditor(UiSignEditInteraction.java:174)
25.07 20:10:26 [Server] INFO at SignEdit-1.14.2.jar//net.deltik.mc.signedit.interactions.UiSignEditInteraction.interact(UiSignEditInteraction.java:153)
25.07 20:10:26 [Server] INFO at SignEdit-1.14.2.jar//net.deltik.mc.signedit.listeners.CoreSignEditListener.overrideNativeBehavior(CoreSignEditListener.java:163)
25.07 20:10:26 [Server] INFO at SignEdit-1.14.2.jar//net.deltik.mc.signedit.listeners.CoreSignEditListener.onRightClickSign(CoreSignEditListener.java:122)
25.07 20:10:26 [Server] INFO at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor734.execute(Unknown Source)
25.07 20:10:26 [Server] INFO at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
25.07 20:10:26 [Server] INFO at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
25.07 20:10:26 [Server] INFO at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
25.07 20:10:26 [Server] INFO at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
25.07 20:10:26 [Server] INFO at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
25.07 20:10:26 [Server] INFO at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
25.07 20:10:26 [Server] INFO at org.bukkit.craftbukkit.v1_20_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:566)
25.07 20:10:26 [Server] INFO at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:537)
25.07 20:10:26 [Server] INFO at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1960)
25.07 20:10:26 [Server] INFO at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37)
25.07 20:10:26 [Server] INFO at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:9)
25.07 20:10:26 [Server] INFO at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$0(PlayerConnectionUtils.java:51)
25.07 20:10:26 [Server] INFO at net.minecraft.server.TickTask.run(TickTask.java:18)
25.07 20:10:26 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
25.07 20:10:26 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1339)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197)
25.07 20:10:26 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1316)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1309)
25.07 20:10:26 [Server] INFO at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.p_(MinecraftServer.java:1287)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1175)
25.07 20:10:26 [Server] INFO at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
25.07 20:10:26 [Server] INFO at java.base/java.lang.Thread.run(Thread.java:833)
25.07 20:10:26 [Server] INFO Caused by: java.lang.IllegalArgumentException: Sign must be in same world as Player
25.07 20:10:26 [Server] INFO at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
25.07 20:10:26 [Server] INFO at org.bukkit.craftbukkit.v1_20_R1.block.CraftSign.openSign(CraftSign.java:127)
25.07 20:10:26 [Server] INFO at org.bukkit.craftbukkit.v1_20_R1.entity.CraftPlayer.openSign(CraftPlayer.java:2759)
25.07 20:10:26 [Server] INFO ... 34 more

**Comment by Deltik** [comment]
This is the real error:

```
25.07 20:10:26 [Server] INFO Caused by: java.lang.IllegalArgumentException: Sign must be in same world as Player
25.07 20:10:26 [Server] INFO at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
25.07 20:10:26 [Server] INFO at org.bukkit.craftbukkit.v1_20_R1.block.CraftSign.openSign(CraftSign.java:127)
25.07 20:10:26 [Server] INFO at org.bukkit.craftbukkit.v1_20_R1.entity.CraftPlayer.openSign(CraftPlayer.java:2759)
```

It means that the player somehow right-clicked a sign in a different world or perhaps another plugin faked a sign right-mouse click on a sign block in a different world.

My best guess is that another plugin is sending bad data to SignEdit for Bukkit.  I can work around this by doing the world match check in SignEdit for Bukkit to ignore a mismatch, which would silence the error.

**Comment by Deltik** [comment]
The fix has landed in [SignEdit for Bukkit v1.14.3](https://github.com/Deltik/SignEdit/releases/tag/v1.14.3) ([direct download](https://github.com/Deltik/SignEdit/releases/download/v1.14.3/SignEdit-1.14.3.jar)).
